lib/active_record_shards/connection_switcher.rb

```diff
@@ -171,6 +174,11 @@ def switch_connection(options)
           options[:replica] ||= options.delete(:slave)
         end
 
+        if options.key?(:master)
+          ActiveRecordShards::Deprecation.warn('the `:master` option should be replaced with `:primary`!')
```

Comment:
This warning is not necessary, since any `:master` option would just be ignored anyway – it isn’t being used in this method.

But looking harder at this change made me realize that a bug was introduced for the `:replica` code – `current_shard_selection.on_replica` is _not_ being set when the option `:slave` is used instead of `:replica`! Changing tests and implementation at the same time _does_ have its downsides 😅